### PR TITLE
Add git repo as a top-level menu

### PR DESCRIPTION
**Pull request by flyrain**
# Description

- [x] Documentation update

# How Has This Been Tested?

Local test works as the following screenshot shows.

Screenshot 2024-10-14 at 11 53 41 AM

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code

### CODE REVIEW

File: site/hugo.yaml [hunk 1]
```diff
@@ -85,9 +85,12 @@ imaging:
 
 menu:
   main:
+    - name: GitHub
+      url: https://github.com/apache/polaris
+
     - name: "Docs & Releases"
       identifier: "releases"
-      weight: 800
+      weight: 100
```
Comment (flyrain) [inline site/hugo.yaml:93]: Make "docs & releases" as the first one.

File: site/layouts/partials/navbar.html [hunk 2]
```diff
@@ -23,100 +23,102 @@
...
+            {{ $children = where .Children.ByWeight "Identifier" "ne" "all-releases-page" }}
+            {{ end }}
+            {{ end }}
+            {{ range $children }}
+            <a class="dropdown-item" href="{{ .URL }}">{{ .Name }}</a>
+            {{ end }}
+          </ul>
+        </div>
+        {{ else }}
+        <!-- Menu items without children -->
```
Comment (flyrain) [inline site/layouts/partials/navbar.html:66]: Add the logic for menu items without children. Other changes in this PR are mainly formatting.

Comment (snazy) [reply site/layouts/partials/navbar.html:66]: Can you remove all the formatting changes? It's a bit hard to see the actual differences.

Comment (flyrain) [reply site/layouts/partials/navbar.html]: Hi @snazy , removed the formatting changes.

**Review by snazy — APPROVED** [review]

**Comment by snazy** [comment]
LGTM! Thanks!

**Pull request merged**